src/Symfony/Component/Serializer/CHANGELOG.md

```diff
@@ -5,6 +5,7 @@ CHANGELOG
 -----
 
  * deprecated the `XmlEncoder::TYPE_CASE_ATTRIBUTES` constant, use `XmlEncoder::TYPE_CAST_ATTRIBUTES` instead
+ * added option to output an UTF-8 BOM in CSV encoder via `CsvEncoder::OUTPUT_BOM` context option
```

Comment:
OUTPUT_UTF8_BOM?
btw, should we ensure the output is really UTF-8 before prepending this?

Reply:
Changed. As for checking whether output is UTF-8 I'd defer this decision to the end user, they'll know best whether they need and want BOM to be included.

Reply:
But adding the UTF-8 BOM when the output is not UTF-8 would be plain wrong.
This should throw an exception to me.

Reply:
What's the preferred way of encoding detection? `mb_detect_encoding($str, 'UTF-8', true)`?

Reply:
All righty, added with a test